### PR TITLE
5615 "add data" button on navbar

### LINKED ISSUE #5615: Add "Add Data" to navbar, plus Host Dataverse input to create dataverse/dataset forms
One way to have people create more dataverses is to make it easier to see how to do it. People who get to dataverse from google and bypass the home page or search results page don't see a clear way to add data. 

One solution: adding a button to the header on every page will help make it clear how to add to Harvard Dataverse. 

**_Adding a button. The button has a dropdown to select a new dataverse or new dataset, as current Add Data functionality from the buttons on the home page:_**
add_data_header_loggedin

If users are not logged in, the Log in/Sign in popup with a link to sign up (issue #6170) is shown. 
Proposed flow:
Add_data_header_flow

**_Removing the "Add Data" button from the search results page, and showing it on the header for not logged in users:_** 
add_data_header_search_nologin

### PR DESCRIPTION
**Pull request by landreev**
## Related Issues

- closes #5615: add "Add Data" button to the navbar

## Pull Request Checklist

- [ ] Unit [tests][] completed
- [ ] Integration [tests][]: None
- [ ] Deployment requirements, [SQL updates][], [Solr updates][], etc.: None
- [ ] [Documentation][docs] completed
- [x] Merged latest from "develop" [branch][] and resolved conflicts

[tests]: http://guides.dataverse.org/en/latest/developers/testing.html
[SQL updates]: http://guides.dataverse.org/en/latest/developers/sql-upgrade-scripts.html
[Solr updates]: https://github.com/IQSS/dataverse/blob/develop/conf/solr/7.3.0/schema.xml
[docs]: http://guides.dataverse.org/en/latest/developers/documentation.html
[branch]: http://guides.dataverse.org/en/latest/developers/branching-strategy.html

### CODE REVIEW

File: doc/sphinx-guides/source/user/dataverse-management.rst [hunk 1]
```diff
@@ -5,40 +5,40 @@ A dataverse is a container for datasets (research data, code, documentation, and
...
+    * **Dataverse Name**: Enter the name of your dataverse.
+    * **Affiliation**: Add any affiliation that can be associated with this particular dataverse (e.g., project name, institute name, department name, journal name, etc). This is automatically filled out if you have added an affiliation for your user account.
     * **Identifier**: This is an abbreviation, usually lower-case, that becomes part of the URL for the new dataverse. Special characters (~,\`, !, @, #, $, %, ^, &, and \*) and spaces are not allowed. **Note**: if you change this field in the future, the URL for your Dataverse will change (http//.../'identifier'), which will break older links to the page.
+    * **Host Dataverse**: Select a dataverse you would like to create the new dataverse in, by default the root dataverse is selected if you click the Add Data button from the root dataverse, your My Data pg, the Add Data link in the navbar, or the Add a dataset button on the custom homepage (if configured), but you can select any dataverse for which you the proper permissions to create dataverses.
```
Comment (scolapasta) [inline doc/sphinx-guides/source/user/dataverse-management.rst]: Should this now be higher since it was moved on the page?

Comment (mheppler) [reply doc/sphinx-guides/source/user/dataverse-management.rst]: Great point. Done.

**Comment by kcondon** [comment]
issues:
-on create dataverse, throws server log error with duplicate jsf id error but works.
-initial validation errors (eg. affiliation blank) is not displayed on create likely due to dupe id error halting further processing but works second time. other validation, eg. category may work.
-changing target dataverse for creating dataverse is ignored, always created in root.
-changing target dataverse for creating dataset throws string parsing error in log and UI remains in grayed out/processing state.

**Comment by landreev** [comment]
I believe the issues listed above have been fixed. Except for the last one from the list (below) - my recollection is that we were unable to reproduce that one, and that the dataset page was actually working properly. It appears to be functioning correctly now. 

> -changing target dataverse for creating dataset throws string parsing error in log and UI remains in grayed out/processing state.

**Pull request merged**